rbc/remotejit.py

```diff
@@ -419,11 +426,17 @@ def __init__(self, host='localhost', port=11532,
           When True, use local client. Useful for debugging.
         debug : bool
           When True, output debug messages.
+        use_tracing_allocator : bool
+          When True, enable the automatic detection of memory leaks.
         """
         if host == 'localhost':
             host = get_local_ip()
 
+        if use_tracing_allocator and not local:
+            raise ValueError('use_tracing_allocator=True can be used only with local=True')
```

Comment:
What restricts using the tracing allocator with a remote client?

Reply:
it's not a fundamental restriction, but for now it's just a missing functionality. Currently it works this way:
1. the `TracingAllocator` records the allocations on the remote side (where the code is exectued)
2. the `LeakDetector` checks whether the `TracingAllocator` leaked anything inside a `with:` statement around a call site, which happens on the local side

To make it working with `local=False`, we would need to implement one or more thrift calls so that the `TracingAllocator` and the `LeakDetector` can communicate. I even started to do it, but then I realized that it was becoming unnecessarily complex and that for the sake of running our own tests `local=True` works well enough and it's much simpler to implement.

Reply:
Thanks! I created https://github.com/xnd-project/rbc/issues/391 to keep track of this limitation. The issue is not high-priority but a good-first-task.